jmh-core/src/main/java/org/openjdk/jmh/profile/XCTraceTableOfContentsHandler.java

```diff
@@ -0,0 +1,118 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.jmh.profile;
+
+import org.xml.sax.Attributes;
+
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.*;
+
+/**
+ * Extracts information about xctrace result tables from trace file's table of contents.
+ *
+ * The most interesting part of the ToC is a set of tables containing information about table's format and
+ * various recoding parameters, such as names of collected PMCs.
+ */
+class XCTraceTableOfContentsHandler extends XCTraceTableHandler {
+    private static final DateTimeFormatter TOC_DATE_FORMAT = DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+    private final List<XCTraceTableDesc> supportedTables = new ArrayList<>();
+
+    private long recordStartMs;
+
+    public List<XCTraceTableDesc> getSupportedTables() {
+        return Collections.unmodifiableList(supportedTables);
+    }
+
+    public long getRecordStartMs() {
+        return recordStartMs;
+    }
+
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes attributes) {
+        setNeedParseCharacters(qName.equals(XCTraceTableHandler.START_DATE));
+        if (!qName.equals(XCTraceTableHandler.TABLE)) {
+            return;
+        }
+
+        String schema = Objects.requireNonNull(attributes.getValue(XCTraceTableHandler.SCHEMA), "Schema not found");
+        if (schema.equals(ProfilingTableType.CPU_PROFILE.tableName)) {
+            supportedTables.add(XCTraceTableDesc.CPU_PROFILE);
+        } else if (schema.equals(ProfilingTableType.TIME_PROFILE.tableName)) {
+            supportedTables.add(XCTraceTableDesc.TIME_PROFILE);
+        } else if (schema.equals(ProfilingTableType.COUNTERS_PROFILE.tableName)) {
+            parseCountersProfile(attributes);
+        }
```

Comment:
Optional: You can use switch over Strings here?

Reply:
It won't work here as loading the enum's field is not a compile-time const expression.

And converting a `schema` value into a `ProfilingTableType` and then switching over it seems to be an overkill here (but if you disagree, I can rewrite it).

Reply:
OK, that's fine. Do we need `} else {` with some exception throwing for exhaustiveness?

Reply:
We don't: TOC-file contains multiple tables, and it's OK if a table we're currently looking at is not listed in `ProfilingTableType`. If there were no supported tables, an exception would be thrown later by the `XCTraceAsmProfiler.`